src/main/java/de/qabel/desktop/ui/actionlog/ActionlogController.java

```diff
@@ -21,75 +31,143 @@
+			Contact contact = contactRepository.findByKeyId(identity, d.getSenderKeyId());
+
+			if (lastDate == null || lastDate.getTime() < d.getCreationDate().getTime()) {
+				lastDate = d.getCreationDate();
+				dropMessageRepository.addMessage(d, contact, false);
+			}
+		}
+		loadMessages(c);
+	}
 
-		injectionContext = new HashMap<>();
+	void sendDropMessage(Contact c, String text) throws QblDropPayloadSizeException, QblNetworkInvalidResponseException, PersistenceException {
+		DropMessage d = new DropMessage(identity, text, "dropMessage");
+		dropMessageRepository.addMessage(d, c, true);
+		httpDropConnector.send(c, d);
+	}
+
+	void loadMessages(Contact c) throws EntityNotFoundExcepion {
+		try {
+			messages.getChildren().clear();
+			List<PersitsDropMessage> dropMessages = dropMessageRepository.loadConversation(c);
+			for (PersitsDropMessage d : dropMessages) {
+
+				if (lastDate == null || lastDate.getTime() < d.getDropMessage().getCreationDate().getTime()) {
+					lastDate = d.getDropMessage().getCreationDate();
+				}
+
+				if (d.getSend()) {
+					addOwnMessageToActionlog(d.getDropMessage());
+				} else {
+					addMessageToActionlog(d.getDropMessage());
+				}
+			}
+		} catch (PersistenceException e) {
+			e.printStackTrace();
+		}
+	}
 
-		OtherTextWrapper wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		Identity i = clientConfiguration.getSelectedIdentity();
-		Contact contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
+	void addMessageToActionlog(DropMessage dropMessage) throws EntityNotFoundExcepion {
+		lastDate = dropMessage.getCreationDate();
+		Map<String, Object> injectionContext = new HashMap<>();
+		Contact sender = contactRepository.findByKeyId(identity, dropMessage.getSenderKeyId());
+		injectionContext.put("dropMessage", dropMessage);
+		injectionContext.put("contact", sender);
 		OtherActionlogItemView otherItemView = new OtherActionlogItemView(injectionContext::get);
 		messages.getChildren().add(otherItemView.getView());
 		messageView.add(otherItemView);
-		text = "short Text";
-		injectionContext = new HashMap<>();
-		injectionContext.put("text", text);
-		myItemView = new MyActionlogItemView(injectionContext::get);
-		messages.getChildren().add(myItemView.getView());
-		messageView.add(myItemView);
+	}
 
-		wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		i = clientConfiguration.getSelectedIdentity();
-		contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
-		otherItemView = new OtherActionlogItemView(injectionContext::get);
-		messages.getChildren().add(otherItemView.getView());
-		messageView.add(otherItemView);
+	void addOwnMessageToActionlog(DropMessage dropMessage) {
 
+		if (dropMessage.getDropPayload() == null || dropMessage.getDropPayload().equals("")) {
```

Comment:
why?
Input validation prevents empty strings and who should create null payload how?